gromacs: Intel oneapi 2023 fixes

danielahlin opened the pull request:
I made two changes to build gromacs with Intel oneapi 2023

1. specify g++ path (and introduced dependency on gcc)
2. Disable Intel MKL discovery mechanism for 2023 as it breaks for 2023 and also seems no longer needed

danielahlin committed:
gromacs: oneapi@2023 - g++-dependency and MKL path

Fixes to build with oneapi@2023 and MKL 2023.
* Depend on gcc and add GMX_GPLUSPLUS_PATH to it
* MKL seems to be picked up without explicit directives
  and the old directives fails (at least for my tests) with MKL
  2023.

danielahlin committed:
Merge branch 'spack:develop' into gromacs-fix-mkl-2023

marvinbernhardt commented:
Sadly, I know next to nothing about oneapi. @danielahlin , since I stopped using Gromacs recently and you seem to be actively working with it, would you be interested in replacing me as a maintainer? I don't know the process for this, and it is likely not my sole decision.

danielahlin committed:
Merge branch 'develop' into gromacs-fix-mkl-2023

danielahlin committed:
gromacs: oneapi@2023 fix - address style errors

marvinbernhardt commented:
> > Sadly, I know next to nothing about oneapi. @danielahlin , since I stopped using Gromacs recently and you seem to be actively working with it, would you be interested in replacing me as a maintainer? I don't know the process for this, and it is likely not my sole decision.
> 
> @marvinbernhardt, I'm happy to help with maintaining the package. However, I want to mention that I'm not an MD subject matter expert nor regular Gromacs user but rather someone who builds and benchmarks Gromacs frequently.

I think that is fine. @alalazo how can we change the maintainer? Just make a PR?

alalazo commented:
@marvinbernhardt Let's just add @danielahlin GitHub user name to the maintainers directive in this PR

danielahlin committed:
gromacs: add danielahlin as maintainer

Adding danielahlin as maintainer as discussed in https://github.com/spack/spack/pull/36555#issuecomment-1496224128 and https://github.com/spack/spack/pull/36555#issuecomment-1504682712

danielahlin commented:
@marvinbernhardt and @alalazo I've added myself as maintainer as discussed.

marvinbernhardt commented on var/spack/repos/builtin/packages/gromacs/package.py:
```diff
@@ -26,7 +26,7 @@ class Gromacs(CMakePackage):
     list_url = "https://ftp.gromacs.org/gromacs"
     git = "https://gitlab.com/gromacs/gromacs.git"
-    maintainers("junghans", "marvinbernhardt")
+    maintainers("danielahlin", "junghans", "marvinbernhardt")
```
suggestion: remove marvinbernhardt

junghans replied:
Done added @eirrgang instead.

junghans committed:
Update package.py

junghans committed:
Merge branch 'develop' into gromacs-fix-mkl-2023

alalazo committed:
Merge branch 'develop' into gromacs-fix-mkl-2023
